consensus/src/experimental/execution_phase.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use channel::{Sender, Receiver};
+use consensus_types::block::Block;
+use diem_infallible::Mutex;
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use execution_correctness::ExecutionCorrectness;
+use executor_types::Error as ExecutionError;
+use consensus_types::executed_block::ExecutedBlock;
+use futures::{StreamExt, SinkExt};
+
+/// [ This class is used when consensus.decoupled = true ]
+/// ExecutionPhase is a singleton that receives ordered blocks from
+/// the ordering state computer and execute them. After the execution is done,
+/// ExecutionPhase sends the ordered blocks to the commit phase.
+pub struct ExecutionPhase {
+    executor_channel_recv: Receiver<(Vec<Block>, LedgerInfoWithSignatures)>,
+    execution_correctness_client: Mutex<Box<dyn ExecutionCorrectness + Send + Sync>>,
```

Comment:
do we need Mutex here?

Reply:
Nice find. Now the execution phase is a singleton. We can remove the mutex.